hipchat/hipchat.go

```diff
@@ -100,6 +116,35 @@ var AuthTest = false
 // API calls if AuthTest=true.
 var AuthTestResponse = map[string]interface{}{}
 
+// RetryOnRateLimit can be set to true to automatically retry the API call until it succeeds,
+// subject to the RateLimitRetryPolicy settings.  This behavior is only active when the API
+// call returns 429 (StatusTooManyRequests).
+var RetryOnRateLimit = false
+
+// RetryPolicy defines a RetryPolicy.
+type RetryPolicy struct {
+	// MaxRetries is the maximum number of attempts to make before returning an error
+	MaxRetries int
+	// MinDelay is the initial delay between attempts.  This value is multiplied by the current attempt number.
+	MinDelay time.Duration
+	// MaxDelay is the largest delay between attempts.
+	MaxDelay time.Duration
+	// JitterDelay is the amount of random jitter to add to the delay.
+	JitterDelay time.Duration
+	// JitterBias is the amount of jitter to remove from the delay.
+	JitterBias time.Duration
+}
+
+// NoRateLimitRetryPolicy defines the "never retry an API call" policy's values.
+var NoRateLimitRetryPolicy = RetryPolicy{0, 1 * time.Second, 1 * time.Second, 500 * time.Millisecond, 0 * time.Millisecond}
+
+// DefaultRateLimitRetryPolicy defines the "up to 300 times, 1 second apart, randomly adding an additional up-to-500 milliseconds of delay" policy.
+var DefaultRateLimitRetryPolicy = RetryPolicy{300, 1 * time.Second, 1 * time.Second, 500 * time.Millisecond, 0 * time.Millisecond}
+
+// RateLimitRetryPolicy can be set to a custom RetryPolicy's values,
+// or to one of the two predefined ones: NoRateLimitRetryPolicy or DefaultRateLimitRetryPolicy
+var RateLimitRetryPolicy = &DefaultRateLimitRetryPolicy
```

Comment:
This implementation was inspired by some other work where the `nil` `RetryPolicy` was effectively the `NO` switch.    I was also prematurely optimizing against memory copies of an inexpensive struct (nanoseconds) in comparison to network delay (milliseconds).  

Will address.